Source/RequestBuilder.swift

```diff
@@ -310,12 +310,37 @@ import AdSupport
         let lmtAd: Bool = !ASIdentifierManager.shared().isAdvertisingTrackingEnabled
         // Limit ad tracking
         deviceDict["lmt"] = NSNumber(value: lmtAd).intValue
-
-        let deviceId = RequestBuilder.DeviceUUID()
-        if deviceId != "" {
-            deviceDict["ifa"] = deviceId
+        
+        //fetch advertising identifier based TCF 2.0 Purpose1 value
+        //truth table
+        /*
+                            deviceAccessConsent=true  deviceAccessConsent=false  deviceAccessConsent undefined
+         gdprApplies=false        Yes, read IDFA       No, don’t read IDFA           Yes, read IDFA
+         gdprApplies=true         Yes, read IDFA       No, don’t read IDFA           No, don’t read IDFA
+         gdprApplies=undefined    Yes, read IDFA       No, don’t read IDFA           Yes, read IDFA
+         */
+          
+        var setDeviceId:Bool = false
+        let purposeConsents:String? = Targeting.shared.purposeConsents
+        
+        if (purposeConsents == nil && Targeting.shared.subjectToGDPR == false) {
```

Comment:
what about the case when  `gdprApplies=undefined` and `deviceAccessConsent undefined` ?

Reply:
This case is handled similar to the gdprApplies = no && deviceConsent = no

Reply:
Yes I see this behaviour in the code, the question in logic:

`gdprApplies = no && deviceConsent = no`  - result: `No, don’t read IDFA`

table above describes 
`gdprApplies = undefined && deviceConsent = undefined` - result: `Yes, read IDFA`